src/foundry/foundry.js/roll.d.ts

```diff
@@ -1,501 +1,515 @@
-   * const plus = new OperatorTerm({operator: "+"});
-   * const t2 = new NumericTerm({number: 8});
-   * const roll = Roll.fromTerms([t1, plus, t2]);
-   * roll.formula; // 4d8 + 8
-   * ```
-   */
-  static fromTerms<D extends Record<string, unknown>>(terms: RollTerm[], options?: Partial<Roll.Options>): Roll<D>;
+    /**
+     * Represent the data of the Roll as an object suitable for JSON serialization.
+     * @returns Structured data which can be serialized into JSON
+     */
+    toJSON(): {
+      class: string;
+      options: Roll.Options;
+      dice: DiceTerm[];
+      formula: string;
+      terms: RollTerm[];
+      total: number | undefined;
+      evaluated: boolean;
+    };
 
-  /**
-   * Expand an inline roll element to display it's contained dice result as a tooltip
-   * @param a - The inline-roll button
-   */
-  static expandInlineResult(a: HTMLAnchorElement): Promise<void>;
+    /**
+     * Recreate a Roll instance using a provided data object
+     * @param data - Unpacked data representing the Roll
+     * @returns A reconstructed Roll instance
+     */
+    static fromData<T extends Roll>(this: ConstructorOf<T>, data: Roll.Data): T;
 
-  /**
-   * Collapse an expanded inline roll to conceal it's tooltip
-   * @param a - The inline-roll button
-   */
-  static collapseInlineResult(a: HTMLAnchorElement): void;
-}
+    /**
+     * Recreate a Roll instance using a provided JSON string
+     * @param json - Serialized JSON data representing the Roll
+     * @returns A reconstructed Roll instance
+     */
+    static fromJSON(json: string): Roll;
 
-declare namespace Roll {
-  // TODO: maybe move this to chat
-  interface ChatOptions {
     /**
-     * @defaultValue `false`
+     * Manually construct a Roll object by providing an explicit set of input terms
+     * @param terms -The array of terms to use as the basis for the Roll
+     * @param options - Additional options passed to the Roll constructor
+     * @returns The constructed Roll instance
+     *
+     * @example
+     * ```typescript
+     * const t1 = new Die({number: 4, faces: 8};
+     * const plus = new OperatorTerm({operator: "+"});
+     * const t2 = new NumericTerm({number: 8});
+     * const roll = Roll.fromTerms([t1, plus, t2]);
+     * roll.formula; // 4d8 + 8
+     * ```
      */
-    blind?: boolean;
+    static fromTerms<D extends Record<string, unknown>>(terms: RollTerm[], options?: Partial<Roll.Options>): Roll<D>;
 
     /**
-     * @defaultValue `null`
+     * Expand an inline roll element to display it's contained dice result as a tooltip
+     * @param a - The inline-roll button
      */
-    flavor?: any;
+    static expandInlineResult(a: HTMLAnchorElement): Promise<void>;
 
-    isPrivate?: boolean;
+    /**
+     * Collapse an expanded inline roll to conceal it's tooltip
+     * @param a - The inline-roll button
+     */
+    static collapseInlineResult(a: HTMLAnchorElement): void;
+  }
 
-    template?: string;
+  namespace Roll {
```

Comment:
Instead of keeping this namespace and making it available globally, consider just moving all of the types declared here out of the `declare global` block.